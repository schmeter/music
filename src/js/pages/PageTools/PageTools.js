import React from 'react';
import PropTypes from 'prop-types';
import { useParams } from 'react-router-dom';

import { Page404 } from '../';
import Page from '../Page';
import Link from '../../components/Link';
import i18n from '../../helpers/i18n';
import Learn from './Learn';
import Clock from './Clock';
import { getRouteWithParams } from '../../helpers/navigation';

const tools = {
  learn: <Learn />,
  clock: <Clock />,
};

const PageTools = ({ closeLayers }) => {
  const { toolId } = useParams();
  const validParams = toolId ? tools[toolId] : true;

  return !validParams ? <Page404 /> : (
    <Page id="tools">
      {toolId ? tools[toolId] : (
        <>
          <h2 className="headline bold">
            <Link to={getRouteWithParams('tools')}>
              {i18n('page_tools_headline')}
            </Link>
          </h2>
          <hr />
          <div className="center">
            {Object.keys(tools).map(key => (
              <p key={key}>
                <Link
                  to={getRouteWithParams('tools', {
                    toolId: key,
                    itemId: '',
                  })}
                  onClick={closeLayers}
                >
                  {i18n(`page_tools_${key}_title`)}
                </Link>
              </p>
            ))}
          </div>
        </>
      )}
    </Page>
  );
};

PageTools.propTypes = {
  closeLayers: PropTypes.func.isRequired,
};

export default PageTools;
