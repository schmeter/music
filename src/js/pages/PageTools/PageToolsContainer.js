import { connect } from 'react-redux';

import PageTools from './PageTools';
import { closeLayersAction } from '../../state/actions/layer';

export const mapDispatchToProps = dispatch => ({
  closeLayers: () => dispatch(closeLayersAction()),
});

export default connect(null, mapDispatchToProps)(PageTools);
