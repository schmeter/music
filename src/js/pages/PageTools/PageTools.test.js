import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PageTools from './PageTools';

let mockUrlParams;

jest.mock('react-router-dom', () => ({
  useParams: () => (mockUrlParams),
}));
// eslint-disable-next-line react/display-name
jest.mock('../../components/Page', () => () => <div />);
// eslint-disable-next-line react/display-name
jest.mock('./Clock', () => () => <div />);
// eslint-disable-next-line react/display-name
jest.mock('./Learn', () => () => <div />);
// eslint-disable-next-line react/display-name
jest.mock('./Watch', () => () => <div />);
jest.mock('../', () => ({
  // eslint-disable-next-line react/display-name
  Page404: () => <div />,
}));

describe('PageTools', () => {
  const props = {
    closeLayers: jest.fn(),
  };

  it('renders correctly without url params', () => {
    mockUrlParams = {};

    const component = shallow(<PageTools {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly with correct url params', () => {
    mockUrlParams = {
      toolId: 'clock',
    };

    const component = shallow(<PageTools {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly with wrong url params', () => {
    mockUrlParams = {
      toolId: 'test',
    };

    const component = shallow(<PageTools {...props} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
