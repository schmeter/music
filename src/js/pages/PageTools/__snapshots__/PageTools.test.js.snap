// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTools renders correctly with correct url params 1`] = `
<Component
  id="tools"
>
  <Component />
</Component>
`;

exports[`PageTools renders correctly with wrong url params 1`] = `<Page404 />`;

exports[`PageTools renders correctly without url params 1`] = `
<Component
  id="tools"
>
  <h2
    className="headline bold"
  >
    <Link
      to="/tools"
    >
      Kann ich helfen?
    </Link>
  </h2>
  <hr />
  <div
    className="center"
  >
    <p
      key="tv"
    >
      <Link
        onClick={[MockFunction]}
        to="/tools/tv"
      >
        Gleich geht's ins Bettchen.
      </Link>
    </p>
    <p
      key="learn"
    >
      <Link
        onClick={[MockFunction]}
        to="/tools/learn"
      >
        Ich wei√ü Bescheid.
      </Link>
    </p>
    <p
      key="clock"
    >
      <Link
        onClick={[MockFunction]}
        to="/tools/clock"
      >
        Keine Zeit.
      </Link>
    </p>
  </div>
</Component>
`;
